spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/tunnel/payload/HttpTunnelPayload.java

```diff
@@ -45,7 +46,7 @@ public class HttpTunnelPayload {
 
 	private static final int BUFFER_SIZE = 1024 * 100;
 
-	protected static final char[] HEX_CHARS = "0123456789ABCDEF".toCharArray();
+	private static final HexFormat HEX_FORMAT = HexFormat.of().withUpperCase();
```

Comment:
This is technically a breaking change if there was someone accessing `HEX_CHARS`, but the chance is very minimal I'd say.